tests/CMakeLists.txt

```diff
@@ -141,7 +153,7 @@ target_link_libraries(mock dl safestr)
 
 add_test(NAME test_open_common_mock_drv
   WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-  COMMAND gtapi -j ${CMAKE_BINARY_DIR}/tests/configuration.json --gtest_filter=*OpenCommonHW*)
+  COMMAND gtapi  ${CMAKE_BINARY_DIR} --gtest_filter="*drv*:-*event_drv_05*:-*nodrv*")
```

Comment:
Instead of combining all filters into one test, can we separate the filters logically into different tests?
Look at the tests/CMakeLists.txt I have in the feature/c++ branch:
https://github.com/OPAE/opae-sdk/blob/feature/c%2B%2B/tests/CMakeLists.txt
I defined a cmake macro to add the tests.

```
macro(add_gtfilter test_name gtfilter is_mock)
    add_test(NAME ${test_name}
        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
        COMMAND $<TARGET_FILE:gtapi> "--gtest_filter=${gtfilter}")
    if(${is_mock})
        set_tests_properties(
            ${test_name}
            PROPERTIES
            ENVIRONMENT "LD_PRELOAD=./lib/libmock.so")
    endif()
endmacro()

add_gtfilter(CxxProperties "CxxProperties*" False)
add_gtfilter(CxxEnum "CxxEnum*" True)
add_gtfilter(CxxOpenClose "CxxOpen*" True)
add_gtfilter(CxxBuffer "CxxBuffer*" True)
add_gtfilter(CxxExcept "CxxExcept*" False)
add_gtfilter(CxxLog "CxxLog*" False)
```

The output looks like this:
![image](https://user-images.githubusercontent.com/22808196/34177714-68bcde4c-e4b9-11e7-9960-1521611de3cf.png)

Reply:
This pull request for adding test cases , next task : adding filter to tests